version: '3.9'

networks:
  parser_project:
    driver: bridge

services:
  api:
    build:
      context: ./api
      dockerfile: Dockerfile
    ports:
      - "8000:8000"
    volumes:
      - ./.env:/fastapi/.env
    environment:
      - PYTHONUNBUFFERED=1
    depends_on:
      - postgres_db
    networks:
      - parser_project

  parser:
    build:
      context: ./parser_python
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
    volumes:
      - ./.env:/parser/.env
    depends_on:
      - postgres_db
    networks:
      - parser_project
  
  postgres_db:
    image: postgres:15
    container_name: postgres_parser_python
    env_file:
      - .env
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres -d postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5
    deploy:
      resources:
        limits:
          cpus: '1'
          memory: 4G
    networks:
      - parser_project
    volumes:
      - ./database/data/postgres:/var/lib/postgresql/data
    ports:
      - "5432:5432"